libs/langchain/langchain/memory/buffer.py

```diff
@@ -19,20 +19,40 @@ def buffer(self) -> Any:
         """String buffer of memory."""
         return self.buffer_as_messages if self.return_messages else self.buffer_as_str
 
-    @property
-    def buffer_as_str(self) -> str:
-        """Exposes the buffer as a string in case return_messages is True."""
+    async def abuffer(self) -> Any:
```

Comment:
@cbornet I suspect that we'll be deprecating these abstractions eventually. They're over-complicated for what they do. Regardless, I think OK adding async methods.

Reply:
If we 're not sure these abstractions will be kept, I think it's better to not add the async version for now. It's harder to remove things than to add them later.
I'll update.

Reply:
Well actually those are used internally by `aload_memory_variables`. Should they be made private with `_` ?
Since it's an implementation and not an interface, it's maybe not so bad to keep them.

Reply:
For public/private, let's match the syntax of any existing sync variants?

---

Regarding the abstraction, I was referring to BaseChatMemory and implementations. 

If I understand correctly, their primary role is to generate a prompt for an LLM / chat model.

We definitely want the functionality, but need to figure out how to simplify the API and make it more intuitive. One possibility is to expose more standalone functions that can operate on messages (e.g., for history trimming based on number of messages, trimming based on # of tokens, summarization etc.)

Reply:
We for sure want to merge the improvements for `BaseChatMessageHistory` -- that abstraction we definitely want/need!